esp/smc/SMCLib/TpCommon.cpp

```diff
@@ -230,6 +231,77 @@ extern TPWRAPPER_API SecAccessFlags getDZFileScopePermissions(IEspContext& conte
     return getDZPathScopePermissions(context, dropZoneName, dropZonePath, dropZoneHost);
 }
 
+static SecAccessFlags getLegacyDZPhysicalPerms(IEspContext& context, const char* dropZoneName, const char* dropZoneHost,
+    const char* dropZoneFile, bool isPath)
+{
+    if (isEmptyString(dropZoneHost) && isEmptyString(dropZoneName))
+        throw makeStringException(ECLWATCH_INVALID_INPUT, "Neither dropzone plane or host specified.");
+
+    Owned<IUserDescriptor> userDesc = createUserDescriptor();
+    userDesc->set(context.queryUserId(), context.queryPassword(), context.querySignature());
+
+    StringBuffer host;
+    if (isEmptyString(dropZoneHost))
+    {
+        Owned<IPropertyTree> plane = getDropZonePlane(dropZoneName);
+        if (getPlaneHost(host, plane, 0))
+            dropZoneHost = host.str();
+        else
+            dropZoneHost = "localhost";
+    }
+
+    CDfsLogicalFileName dlfn;
+    SocketEndpoint ep(dropZoneHost);
+    dlfn.setExternal(ep, dropZoneFile);
+
+    StringBuffer scopes;
+    const char *scopesToCheck = nullptr;
+    if (isPath)
+        scopesToCheck = dlfn.get();
+    else
+    {
+        dlfn.getScopes(scopes);
+        scopesToCheck = scopes;
+    }
+    return queryDistributedFileDirectory().getFScopePermissions(scopesToCheck, userDesc);
+}
+
+//Get dropzone file permission when the file path does not contain a file name. To get the permission,
+//this function (and the validate functions below) calls the getDZPathScopePermissions() which handles
+//isDropZoneRestrictionEnabled if a dropzone cannot be found based on host and path. 
+extern TPWRAPPER_API SecAccessFlags getDZPathScopePermsAndLegacyPhysicalPerms(IEspContext &context, const char *dropZoneName, const char *path,
+    const char *host, SecAccessFlags permissionReq)
+{
+    SecAccessFlags permission = getDZPathScopePermissions(context, dropZoneName, path, host);
+    if ((permission < permissionReq) && getGlobalConfigSP()->getPropBool("expert/@failOverToLegacyPhysicalPerms", !isContainerized()))
+        permission = getLegacyDZPhysicalPerms(context, dropZoneName, host, path, true);
+    return permission;
+}
+
+//Validate dropzone host and file path. Also validate dropzone file permission when the file path does not contain a file name.
```

Comment:
The problem is inside the validateDropZoneHostAndPath() which is called by validateDropZoneReq() and not check isDropZoneRestrictionEnabled(). It should be fixed because the validateDropZoneHostAndPath() is also called by the FileSpray.FileList. I am working on it now.